src/main/java/games/strategy/engine/lobby/server/login/RsaAuthenticator.java

```diff
@@ -44,48 +37,40 @@
   @VisibleForTesting
   static final String RSA_PUBLIC_KEY = "RSAPUBLICKEY";
 
-  private final Cache<String, PrivateKey> rsaKeyCache;
+  private final KeyPair keyPair;
 
   RsaAuthenticator() {
-    this(CacheBuilder.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).build());
+    this(generateKeyPair());
   }
 
   @VisibleForTesting
-  RsaAuthenticator(final Cache<String, PrivateKey> rsaKeyCache) {
-    this.rsaKeyCache = rsaKeyCache;
+  RsaAuthenticator(final KeyPair keyPair) {
+    this.keyPair = keyPair;
   }
 
   /**
    * Returns true if the specified map contains the required values.
    */
   static boolean canProcessResponse(final Map<String, String> response) {
-    return response.get(ENCRYPTED_PASSWORD_KEY) != null;
+    return response.containsKey(ENCRYPTED_PASSWORD_KEY);
   }
 
   /**
    * Returns true if the specified map contains the required values.
    */
   public static boolean canProcessChallenge(final Map<String, String> challenge) {
-    return challenge.get(RSA_PUBLIC_KEY) != null;
+    return challenge.containsKey(RSA_PUBLIC_KEY);
   }
 
   /**
-   * Adds public key of a generated key-pair to the challenge map
-   * and stores the private key in a map.
+   * Creates a new challenge for the lobby server to send to the lobby client.
+   *
+   * @return The challenge as a collection of properties to be added to the message the lobby server sends the lobby
+   *         client.
    */
-  Map<String, String> generatePublicKey() {
-    return Collections.singletonMap(RSA_PUBLIC_KEY, storeKeyPair());
-  }
-
-  private String storeKeyPair() {
-    KeyPair keyPair;
-    String publicKey;
-    do {
-      keyPair = generateKeyPair();
-      publicKey = Base64.getEncoder().encodeToString(keyPair.getPublic().getEncoded());
-    } while (rsaKeyCache.getIfPresent(publicKey) != null);
```

Comment:
I'm curious why this was in a `while` loop previously. I think I agree that the `do` part of the loop should only be executed once, but it still seems really odd. I really wonder if I'm missing something and the loop really had a reason.

Reply:
I _think_ it was just in case `generateKeyPair()` happened to generate a duplicate key pair that had previously been generated for a different client.  That seems highly unlikely given how RSA key generation works.

@RoiEXLab can you please confirm what your intention was with this loop?

Reply:
Yes this was the reasoning.
It is highly unlikely, but in case it happened a single time it would have prevented a player from logging in once and the player would need to login again...

Reply:
Excellent, thanks. Indeed it sounds like we only need to generate a key pair once, in which case the duplicate key check (the `while` loop) would no longer be necessary.